tests/test_meta.py

```diff
@@ -0,0 +1,38 @@
+# This file is dual licensed under the terms of the Apache License, Version
+# 2.0, and the BSD License. See the LICENSE file in the root of this repository
+# for complete details.
+
+import os
+import pkgutil
+import subprocess
+import sys
+import typing
+
+import cryptography
+
+
+def find_all_modules() -> typing.List[str]:
+    return sorted(
+        mod
+        for _, mod, _ in pkgutil.walk_packages(
+            cryptography.__path__,  # type: ignore[attr-defined]
+            prefix=cryptography.__name__ + ".",
+        )
+    )
+
+
+def test_no_circular_imports(subtests):
+    env = os.environ.copy()
+    env["PYTHONPATH"] = os.pathsep.join(sys.path)
+
+    # When using pytest-cov it attempts to instrument subprocesses. This
+    # causes the memleak tests to raise exceptions.
+    # we don't need coverage so we remove the env vars.
+    env.pop("COV_CORE_CONFIG", None)
+    env.pop("COV_CORE_DATAFILE", None)
+    env.pop("COV_CORE_SOURCE", None)
+
+    for module in find_all_modules():
+        with subtests.test():
+            argv = [sys.executable, "-c", f"__import__({module!r})"]
```

Comment:
Was it a conscious decision not to add `-W error`?

Reply:
Yes. The goal of the test in pytest was to check for warnings, that wasn't our goal here :-)

We in fact _do_ have modules that emit warnings.

Reply:
> We in fact _do_ have modules that emit warnings.

This is why in my implementation `-W error` is followed by a specific `-W ignore` :) But I get why you want to opt-out of this check...

Maybe you'd consider summarizing the considerations in a docstring?

Reply:
I'm going to merge this but if you'd like to add a comment to this effect I'd be happy to review that PR too!